lib/ret/media_resolver.ex

```diff
@@ -191,31 +191,109 @@ defmodule Ret.MediaResolver do
-        # but our client right now will be confused by that because some images need to turn into
-        # image-like views and some (GIFs) need to turn into video-like views.
-        resp ->
-          case resp.body |> OpenGraph.parse() do
-            %{video: video} when is_binary(video) ->
-              [URI.parse(video), %{expected_content_type: "video/*"}]
-
-            # don't send image/*
-            %{image: image} when is_binary(image) ->
-              [URI.parse(image), %{}]
-
-            _ ->
-              [uri, %{expected_content_type: content_type_from_headers(resp.headers)}]
+  defp resolve_non_video(%URI{host: host} = uri, _root_host) do
+    photomnemonic_endpoint = module_config(:photomnemonic_endpoint)
+
+    # Crawl og tags for hubs rooms + scenes
+    is_local_url = host === RetWeb.Endpoint.host()
+
+    # For text/html pages we use the screenshotter, otherwise return the raw URL.
+    # Try HEAD, if HEAD fails then do a GET and check content type.
+    # If HEAD works, check content type.
+    case uri |> URI.to_string() |> retry_head_until_success() do
+      :error ->
+        case uri |> URI.to_string() |> retry_get_until_success([{"Range", "bytes=0-32768"}]) do
+          %HTTPoison.Response{headers: headers} = res ->
+            content_type = headers |> content_type_from_headers
+
+            if !is_local_url && photomnemonic_endpoint && content_type |> String.starts_with?("text/html") do
+              screenshot_commit_for_uri(uri, content_type)
+            else
+              og_tag_commit_for_response(uri, res)
+            end
+
+          :error ->
+            nil
+        end
+
+      %HTTPoison.Response{headers: headers} ->
+        content_type = headers |> content_type_from_headers
+
+        if !is_local_url && photomnemonic_endpoint && content_type |> String.starts_with?("text/html") do
+          screenshot_commit_for_uri(uri, content_type)
+        else
+          case uri |> URI.to_string() |> retry_get_until_success([{"Range", "bytes=0-32768"}]) do
+            :error ->
+              nil
+
+            resp ->
+              og_tag_commit_for_response(uri, resp)
           end
+        end
+    end
+  end
+
+  defp screenshot_commit_for_uri(uri, content_type) do
+    photomnemonic_endpoint = module_config(:photomnemonic_endpoint)
+    photomnemonic_api_id = module_config(:photomnemonic_api_id)
+
+    query = URI.encode_query(url: uri |> URI.to_string())
+
+    cached_file_result =
+      CachedFile.fetch("screenshot-#{query}", fn path ->
+        Statix.increment("ret.media_resolver.screenshot.requests")
+
+        headers =
+          if photomnemonic_api_id do
+            %{"x-amzn-apigateway-api-id" => photomnemonic_api_id}
+          else
+            %{}
+          end
+
+        url = "#{photomnemonic_endpoint}/screenshot?#{query}"
+        {:ok, _path} = Download.from(url, path: path, headers: headers)
```

Comment:
Don't think we handle the case where this fails. I think the intent is for it to fall back to fetching OG tags.